# thesis
thesis-work figures 





***

![.Figure1.1/figure1.1.png](https://raw.githubusercontent.com/TWarczak/thesis/master/Figure1.1/figure1.1.png)

**Figure 1.1: Generalized Diagram of Arsenic Transport and Metabolism in Plants**
Modified from Punshon et al. (2017), Zhao et al. (2009), and Ma et al. (2007). 

***

![.Ch_2_figures/figure2.1A_B_C.png](https://raw.githubusercontent.com/TWarczak/thesis/master/Ch_2_figures/figure2.1A_B_C.png)

![.Ch_2_figures/figure2.1D.png](https://raw.githubusercontent.com/TWarczak/thesis/master/Ch_2_figures/figure2.1D.png)

**Figure 2.1: Unsupervised Clustering & Exploratory Data Analysis**
(A) Principal Component Analysis (PCA) plot showing the samples clustering based on the two principal components, cell-type and condition, which explains 14% and 79% of the variation, respectively, in the RNA-seq data.  (B) Hierarchical clustering heatmap showing similarities in global gene expression between both the biological replicates and the different sample groups.  Perfect correlation = 1.  (C) Heatmap of genes in control conditions that show cell-type specific expression in Dinneney et al. 2008 and this study.  The promoters of At5g14750, At3g54220, and At1g09750 were used to drive expression of GFP in the WER::GFP,  SCR::GFP, and Cortex::GFP marker lines.  Color scale based on difference between RPKM of gene in cell-type and mean RPKM of gene in all samples, divided by the standard deviation (x - mean(x)) / sd(x). (D) Volcano plots showing the fold changes relative to the adjusted p-values for all genes in epidermal, cortex, and endodermal cell-types.  Blue dots are statistically significant differentially expressed genes padj < 0.05, red dots are not significant.  

***

![.Ch_2_figures/figure2.2A_B_C.png](https://raw.githubusercontent.com/TWarczak/thesis/master/Ch_2_figures/figure2.2A_B_C.png)

**Figure 2.2: Differentially Expressed Genes by Cell-Type**
(A) Hierarchical clustering of 7543 significant differentially expressed genes with abs(logFC) > 1 and RPKM > 6 in at least one cell-type of any condition.  8 clusters can be seen showing relative DEG expression in each cell-type and condition.  Color scale the difference between RPKM of gene in cell-type and mean RPKM of gene in all samples, divided by the standard deviation (x - mean(x)) / sd(x).  (B) Venn diagram of same 7543 DEGs differentially expressed in unique or combinations of cell-types.  (C) Density plots of the up (bottom row) and down (top row) DEG RPKM values in epidermal, cortex, or endodermal cells in either control (blue) or AsIII (red) conditions.  

***

![.Ch_2_figures/figure2.2B.png](https://raw.githubusercontent.com/TWarczak/thesis/master/Ch_2_figures/figure2.2B.png)

Figure 2.2B:  

***
